src/ProfileBundle/Controller/ProfileController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ProfileBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * This class is inspired from the FOS Profile Controller, except :
+ *   - only twig is supported
+ *   - separation of the user authentication form with the profile form.
+ */
+class ProfileController extends Controller
```

Comment:
Please make everything new final by default or use private properties where possible

Reply:
It is not new. I copy it from SonataUserBundle 3.x If someone override it when use SonataUserBundle 3.x, then can simple override it by change UserBundle to ProfileBundle